sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExecutionPlanCaptureCallback.scala

```diff
@@ -93,10 +93,13 @@ object ExecutionPlanCaptureCallback extends AdaptiveSparkPlanHelper {
     import org.apache.spark.sql.types.StructType
     import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
 
-    val cpuFileSourceScanSchemata = collect(cpuDf.queryExecution.executedPlan) {
+    val adaptiveSparkPlanHelper = ShimLoader.newAdaptiveSparkPlanHelperShim()
```

Comment:
I think all we need is this, and we can get rid of the changes to ShimLoader and we don't  need the impl class
```suggestion
  val adaptiveSparkPlanHelper = new AdaptiveSparkPlanHelper {}
```